js/src/Debug/Toolbar.vue

```diff
@@ -0,0 +1,445 @@
+<script setup>
+    /**
+     * @typedef {{ id: string, parent_id: string|null, category: string, name: string, start: number, end: number }} ProfilerSection
+     * @typedef {{ execution_time: number, memory_usage: number, memory_peak: number, memory_limit: number }} ServerPerformanceMetrics
+     * @typedef {{ total_requests: number, total_duration: number, queries: { request_id: string, num: number, query: string, time: number, rows: number, warnings: string[], errors: string[] } }} SQLMetrics
+     * @typedef {{ id: string, parent_id: string, server_performance: ServerPerformanceMetrics, sql: SQLMetrics, globals: Object.<string, string>, [profiler]: ProfilerSection[] }} Profile
+     * @typedef {{ id: string, data: Object.<string, string>, url: string, server_global: Object.<string, string>, type: string, start: Date, time: number, status: number, status_type: string, profile: Profile|null }} AJAXRequestData
+     * @typedef {{ x: number, y: number, width: number, height: number }} ClientTimingBounds
+     * @typedef {{ queued: Array, redirect: Array, fetch: Array, dns: Array, connection: Array, initial_connection: Array, ssl: Array, request: Array, response: Array }} ClientTimingSections
+     * @typedef {{ type: string, name: string, start: number, end: number, bounds: Object.<string, ClientTimingBounds>, sections: ClientTimingSections }} ClientTimingData
+     * @typedef {{ id: string, show: function(content_area: jQuery, refresh: boolean) }} Widget
+     */
+    /**
+     * @typedef MainWidget
+     * @extends Widget
+     * @property {boolean} main_widget=true
+     * @property {string} title
+     * @property {string|null} icon
+     * @property {function(button: jQuery)} refreshButton
+     */
+    /**
+     * @typedef SubWidget
+     * @extends Widget
+     * @property {boolean} main_widget=false
+     */
+    import {computed, ref} from "vue";
+
+    const props = defineProps({
+        initial_request: {
+            type: Object,
+            default: null,
+        },
+    });
+
+    const active_widget = ref(null);
+    const active_widget_component = computed(() => {
+        if (active_widget.value === null) {
+            return null;
+        }
+        const widget = widgets.find(w => w.id === active_widget.value);
+        if (widget === undefined) {
+            return null;
+        }
+        return widget.component_registered_name;
+    });
+
+    const ajax_requests = ref([]);
+
+    const current_request = ref(props.initial_request.id);
+    const current_profile = computed(() => {
```

Comment:
This is a computed value. Vue will track all the refs used inside the function and should only update the value returned if a ref changes.